src/engine/optimizer/statistics.rs

```
@@ -0,0 +1,156 @@
+    }
+
+    /// DELETE 반영: 캐시된 통계가 있으면 행 개수와 블록 개수를 증분 갱신합니다.
+    pub async fn record_delete(&self, table: &TableName, count: usize) {
+        if let Some(statistics) = self.statistics.write().await.get_mut(table) {
+            let old_row_count = statistics.row_count;
+            statistics.row_count = statistics.row_count.saturating_sub(count);
+
+            // 블록 개수를 비례적으로 감소
+            if old_row_count > 0 {
+                let avg_rows_per_block = old_row_count / statistics.block_count.max(1);
+                if avg_rows_per_block > 0 {
```

Clippy (1.96.0)

[warning] manual checked division

(warning)